fetcher-plugins/rocketmq-fetcher-plugin/src/main/java/org/skyapm/transporter/fetcher/rocketmq/RocketmqFetcherHandlerRegister.java

```diff
@@ -0,0 +1,155 @@
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Configuring and initializing a Rocketmq Consumer client as a dispatcher to delivery  Message to registered handler by topic.
+ */
+@Slf4j
+public class RocketmqFetcherHandlerRegister implements Runnable {
+
+    private Map<String, RocketmqHandler> handlerMap = new ConcurrentHashMap<>();
+    private DefaultLitePullConsumer consumer;
+    private final RocketmqFetcherConfig config;
+    private int threadPoolSize = Runtime.getRuntime().availableProcessors() * 2;
+    private int threadPoolQueueSize = 10000;
+    private final ThreadPoolExecutor executor;
+    private DefaultMQAdminExt defaultMQAdminExt;
+
+    public RocketmqFetcherHandlerRegister(RocketmqFetcherConfig config) {
+        this.config = config;
+
+        if (config.getRocketmqHandlerThreadPoolSize() > 0) {
+            threadPoolSize = config.getRocketmqHandlerThreadPoolSize();
+        }
+        if (config.getRocketmqHandlerThreadPoolQueueSize() > 0) {
+            threadPoolQueueSize = config.getRocketmqHandlerThreadPoolQueueSize();
+        }
+
+        consumer = new DefaultLitePullConsumer(config.getConsumerGroup());
+        consumer.setNamesrvAddr(config.getNameServers());
+
+        executor = new ThreadPoolExecutor(threadPoolSize, threadPoolSize,
+                60, TimeUnit.SECONDS,
+                new ArrayBlockingQueue<>(threadPoolQueueSize),
+                new CustomThreadFactory("RocketmqConsumer"),
+                new ThreadPoolExecutor.CallerRunsPolicy()
+        );
+    }
+
+    public void register(RocketmqHandler handler) {
+        handlerMap.put(handler.getTopic(), handler);
+    }
+
+    public void start() throws ModuleStartException {
+
+        createTopicIfNeeded(handlerMap.keySet());
+        handlerMap.keySet().forEach(topic -> {
+            try {
+                consumer.subscribe(topic, "*");
+            } catch (MQClientException e) {
+                log.error("skywalking consumer subscribe topic {} fail", topic);
+                e.printStackTrace();
+            }
+        });
+        consumer.setPullBatchSize(config.getPullBatchSize());
+        consumer.setPollTimeoutMillis(config.getPollTimeoutMillis());
+
+        try {
+            consumer.start();
+        } catch (MQClientException e) {
+            log.error("rocketmq consumer start failed");
+            e.printStackTrace();
+        }
+        executor.submit(this);
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            try {
+                List<MessageExt> msgs = consumer.poll();
+                if (msgs.isEmpty()) {
+                    continue;
+                }
+                msgs.forEach(msg -> handlerMap.get(msg.getTopic()).handle(msg));
+                consumer.commitSync();
+            } catch (Exception e) {
+                log.error("rocketmq handle message error", e);
+                e.printStackTrace();
```

Comment:
You have logged the exception in the previous line, no need to print the stack trace again